### PR TITLE
CAMEL-19584 : add KinesisAsyncClient to the camel component

### PR DESCRIPTION
**Pull request by hamed-hatami**
# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

File: components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java [hunk 1]
```diff
@@ -67,6 +67,8 @@ public class Kinesis2Configuration implements Cloneable {
     private Integer proxyPort;
     @UriParam(defaultValue = "false", description = "If we want to trust all certificates in case of overriding the endpoint")
     private boolean trustAllCertificates;
+    @UriParam(defaultValue = "false", description = "If we want to a KinesisAsyncClient instance set it to true")
```
Comment (oscerd) [inline components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java:70]: This is producer only, so it needs to be labelled

File: components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/client/KinesisClientFactory.java [hunk 2]
```diff
@@ -44,4 +45,14 @@ public static KinesisInternalClient getKinesisClient(Kinesis2Configuration confi
...
         }
     }
+
+    /**
+     * Return the standard aws Kinesis Async client.
+     *
+     * @param  configuration configuration
+     * @return               KinesisAsyncClient
+     */
+    public static KinesisAsyncInternalClient getKinesisAsyncClient(Kinesis2Configuration configuration) {
```
Comment (oscerd) [inline components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/client/KinesisClientFactory.java:55]: The default and profile credentials are missing. It should be the same as the Sync client, so you should provide the other two implementations

File: components/camel-aws/camel-aws2-kinesis/pom.xml [hunk 3]
```diff
@@ -33,6 +33,7 @@
     <description>A Camel Amazon Kinesis Web Service Component Version 2</description>
 
     <properties>
+        <amazon-kinesis-common-version>2.5.1</amazon-kinesis-common-version>
```
Comment (oscerd) [inline components/camel-aws/camel-aws2-kinesis/pom.xml:36]: This should go in Parent POM

**Review by oscerd — CHANGES_REQUESTED** [review]
Looks good overall, but I would wait for the consumer to support the async client before merging.

File: components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java (hunk 1 above)
Comment (hamed-hatami) [reply components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java:70]: My aim was to make it bi-directional, both consumer and producer will take effect by async connection ?

Comment (hamed-hatami) [reply components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Configuration.java:70]: Let me declare a label for consumer then  for the time being

File: components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/client/KinesisClientFactory.java (hunk 2 above)
Comment (hamed-hatami) [reply components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/client/KinesisClientFactory.java:55]: It's done

File: components/camel-aws/camel-aws2-kinesis/pom.xml (hunk 3 above)
Comment (hamed-hatami) [reply components/camel-aws/camel-aws2-kinesis/pom.xml:36]: It's also done

**Review by oscerd — APPROVED** [review]

**Pull request merged**